weave

```diff
@@ -1478,6 +1564,13 @@ case "$COMMAND" in
                 --rewrite-hosts)
                     REWRITE_HOSTS=1
                     ;;
+                --add-host)
+                    DNS_EXTRA_HOSTS="$2 $DNS_EXTRA_HOSTS"
+                    shift
+                    ;;
+                --add-host=*)
+                    DNS_EXTRA_HOSTS="${1#*=} $DNS_EXTRA_HOSTS"
+                    ;;
```

Comment:
Instead of doing this here, it would be neater do simply invoke `dns_extra_hosts_args "$@"` in the `-n "$REWRITE_HOSTS` branch below.